transition-scripts/deploy.sh

```diff
@@ -26,18 +26,30 @@ if [ "$#" -lt 1 ]; then
 fi
 
 namespace=$1
+purge="false"
 
 pwd="$(dirname "$0")"
 source "$pwd/helpers/_all.sh"
 
-# TODO: take an argument to just upgrade; no uninstallation, and no cleanup to ensure gold is running as active and golddr is running as standby
+while [[ "$2" =~ ^- && ! "$2" == "--" ]]; do
```

Comment:
I'd just like to confirm my understanding of the logic here:
The first check seems to ensure the $2 argument script starts with a dash, but isn't "--", 
The following logic checks if the the $2 argument is "--purge"  or "-p"
The shift command lets this loop run from the 3rd argument until it hits an arg that does not start with a '-'.  Is that correct?

Reply:
excellent thank you